tuf/ngclient/updater.py

```diff
@@ -307,21 +390,33 @@ def _load_root(self) -> None:
         lower_bound = self._trusted_set.root.signed.version + 1
         upper_bound = lower_bound + self.config.max_root_rotations
 
-        for next_version in range(lower_bound, upper_bound):
-            try:
-                data = self._download_metadata(
-                    Root.type,
-                    self.config.root_max_length,
-                    next_version,
-                )
-                self._trusted_set.update_root(data)
-                self._persist_metadata(Root.type, data)
+        save_repo_version_dir = self._spec_version_dir
 
-            except exceptions.DownloadHTTPError as exception:
-                if exception.status_code not in {403, 404}:
-                    raise
-                # 404/403 means current root is newest available
-                break
+        for next_version in range(lower_bound, upper_bound):
+            for version_repo in supported_version_repos:
```

Comment:
I think that as-written this would accept `/1.root.json`, `/2/2.root.json`, `3.root.json` which feels wrong.

Reply:
Maybe we want to say:

```python
for next_version in range(lower_bound, upper_bound):
    while supported_version_repos:
        if it's in supported_version_repos[0]:
            handle it!
            break
        else:
            supported_version_repos.pop()
```

Reply:
Just added a test, this does ignore `3.root.json` in your example. The complication is that both the `supported_version_repos` look and the `next_version` loop have to break early in some (different) cases, which is where the additional complexity comes in. I'm happy to simplify if there's an easier way to represent this.